composeApp/src/commonMain/kotlin/com/majotyler/hiittimer/presentation/TimerViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.majotyler.hiittimer.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class TimerViewModel : ViewModel() {
+
+    private val _loading = MutableStateFlow(false)
+    val loading = _loading.asStateFlow()
+
+    private val _exercises = MutableStateFlow(listOf("Lagartijas","gallitos", "dominadas"," sentadillas"))
+    val exercises = _exercises.asStateFlow()
+
+    fun onEvent(event: TimerViewEvent) {
+        when (event) {
+            is TimerViewEvent.ClickAdd -> {
+                viewModelScope.launch {
+                    ClickAdd()
+                }}
+            is TimerViewEvent.onDelete -> {
+                onDelete(event.item)
+                }
```

Comment:
Recommended style change :) 

```suggestion
            is TimerViewEvent.ClickAdd -> onClickedAdd()
            is TimerViewEvent.ClickedDelete -> onClickedDelete(event = event)
```

Reply:
The coroutine is necessary there. Or you recommend me put that in the view ? Let me know